src/applications/bmqbrkr/run

```diff
@@ -7,34 +7,43 @@ set -e
 # If 'gdb' or 'lldb' is passed in, execute the task with those arguments in the
 # corresponding debugger.
 
-if [ ! -s 'bmqbrkr.tsk' ]; then
-    echo "Task 'bmqbrkr.tsk' doesn't exist"
-    exit 1
-fi
-
-BMQ_PREFIX="${BMQ_PREFIX:-localBMQ}"
-
-if [ ! -d $BMQ_PREFIX ]; then
-    mkdir $BMQ_PREFIX
-    ln -sf $(readlink etc) $BMQ_PREFIX/etc
-    mkdir $BMQ_PREFIX/logs
-    mkdir -p $BMQ_PREFIX/storage/local/archive
-else
-    echo "Clearing logs ... in $BMQ_PREFIX/logs/*"
-    rm -f $BMQ_PREFIX/logs/*
-    mkdir -p $BMQ_PREFIX/storage/local/archive
-fi
-
-while [ "$1" != "" ]; do
-    if [ "$1" == "gdb" ]; then
-        debugCmd="gdb --args"
-    elif [ "$1" == "lldb" ]; then
-        debugCmd="lldb --"
-    elif [ "$1" == "echo" ]; then
-        debugCmd="echo"
+(
+    # The configuration uses paths relative to this directory for the storage,
+    # because we cannot assume where the repository is located. Thus we need to
+    # 'cd' here, but we do it inb a sub-shell, so the effect is limited to this
+    # script.
+
+    cd $(dirname $0)
+
+    if [ ! -s 'bmqbrkr.tsk' ]; then
+        echo "Task 'bmqbrkr.tsk' doesn't exist"
+        exit 1
+    fi
+
+    BMQ_PREFIX="${BMQ_PREFIX:-localBMQ}"
+
+    if [ ! -d "$BMQ_PREFIX" ]; then
+        mkdir "$BMQ_PREFIX"
```

Comment:
If we get passed `${BMQ_PREFIX}` where the full path doesn't exist (i.e., `a/b/`, where both `a` and `b` don't exist) should we fail or make both `a` and `b`?

`mkdir` or `mkdir -p` in other words?

Reply:
I'm leaning just `mkdir`, but no strong feeling about it.

Reply:
`mkdir` feels a bit safer, just thought I'd bring it up.  Let's go with that.